common-npm-packages/packaging-common/util.ts

```diff
@@ -102,6 +116,86 @@ export enum LogType {
     error
 }
 
+export function getAccessToken(nugetFeedType:string, endpointInputKey:string, feedInputKey:string, packageToolType: PackageToolType) : string {
+    let accessToken: string = getSystemAccessToken();
+    if (nugetFeedType.toLowerCase() == "internal") {
+        let feed = getProjectAndFeedIdFromInputParam(feedInputKey);
+        accessToken = getUserAccessToken(tl.getInput(endpointInputKey), packageToolType, feed);
+    }
+
+    if (!accessToken) {
+        tl.debug(`Access token not set. Using System Access token.`);
+        return getSystemAccessToken();
+    }
+    return accessToken;
+}
+
+function getUserAccessToken(endpointName: string, packageToolType: PackageToolType, feed: any): string {
+    let token = "";
+    tl.debug("Checking if the token is already set.");
+
+    if(endpointName) {
+        tl.debug(`Checking if the endpoint ${endpointName} provided by user, can be used.`);
+        token = getAccessTokenFromServiceConnectionForInternalFeeds(endpointName);
```

Comment:
What type of service connections would be supported by these changes? If we are targeting nuget service connections, will all 3 types be supported here?